web/editor_undo_bar.js

```diff
@@ -39,10 +39,7 @@ class EditorUndoBar {
 
     // Caveat: we have to pick between registering these everytime the bar is
     // shown and not having the ability to cleanup using AbortController.
-    const boundHide = this.hide.bind(this);
-    closeButton.addEventListener("click", boundHide);
-    eventBus._on("beforeprint", boundHide);
-    eventBus._on("download", boundHide);
```

Comment:
Removing these listeners means that a number of ways in which printing and downloading can be triggered will no longer hide the undo-bar.

Reply:
Ctrl+P and Ctrl+S are handled by the key press handler. Are there other ways that involve neither the keyboard nor the toolbar?

Reply:
Assuming that https://github.com/mozilla/pdf.js/pull/18900#discussion_r1806759172 gets done @ryzokuken

Reply:
Nevermind, clicking on the "print" icon in the toolbar does indeed not close the bar.

Reply:
ah shucks atm only clicking on the secondary toolbar dismisses the undo bar. What should I do? Re-add the event listeners or also make the bar go away when someone clicks the main toolbar?

Reply:
After playing around with it, it seems like when using the built-in viewer in firefox (but not when using https://mozilla.github.io/pdf.js/web/viewer.html) you can also save and print using the browser's menu.

I think it's thus better to keep these two event listeners.

Reply:
re-added, thanks both